firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -658,6 +770,57 @@ public void onCancelled(DatabaseError error) {}
     assertEquals("b", removed.get(0));
   }
 
+  @Test
+  public void setLimitEnsureChildRemovedAndChildAddedHitWhenLimitIsHitFromFrontWithStartAfter()
```

Comment:
Totally optional:
s/WhenLimitIsHitFromFront/WhenDataEntersLimit

Here and below if you do update.